### ISSUE TITLE
Could not find SDK "Microsoft.VCLibs.Desktop, Version=14.0".

### ISSUE THREAD

**Issue by SwimmingTiger** [microsoft/terminal#837]
### Your Windows build number
Windows 10 version 1903 build 18895.1000

### What you're doing and what's happening
Got this error:
`Error	MSB3774	Could not find SDK "Microsoft.VCLibs.Desktop, Version=14.0".	OpenConsolePackage	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\Microsoft\DesktopBridge\Microsoft.DesktopBridge.targets	266`

I have exhausted all possible solutions:
* Build with VS2017 (the same error)
* Build with VS2019 Preview 3 (the same error)
* Retarget platform toolset to v142 (the same error)
* Install any components that may be missing (the same error)
* Install the VS2015 platform toolset (v140) (the same error)
* Switch system language from Chinese to English (the same error)
* Switch system code page to UTF-8 (beta) (the same error)
* Switch system code page to English-US (the same error)
* Clone and rebuild many times (the same error)
* Modify any possible version settings (the same error)

![Screenshot (1)](https://user-images.githubusercontent.com/4986069/57814239-a69b1000-77a5-11e9-9019-5e26565d2078.png)

### What's wrong / what should be happening instead

Why does a VS2017 (v141) project require VCLibs 14.0 (v140)? And the error message points to the file from Visual Studio itself. I think this is beyond my understanding.

Wish a Visual Studio expert will tell me what happened.

I think this is a problem specific to this project. I can build a UWP Demo project normally. I can also build the earlier commits (https://github.com/microsoft/Terminal/commit/58ec47236df349e1ed65d47478773035736e36b5 and earlier).

![Screenshot (2)](https://user-images.githubusercontent.com/4986069/57814874-adc31d80-77a7-11e9-87e5-e1427b202d4c.png)

**Comment by SwimmingTiger** [comment]
Try installing VS2015 but it requires 8GB of disk space with C/C++ support. I don't think this is a reasonable solution.

![Annotation 2019-05-16 070650](https://user-images.githubusercontent.com/4986069/57815242-355d5c00-77a9-11e9-8245-0af8de82fce0.jpg)

**Comment by SwimmingTiger** [comment]
It seems that VCLib 14.0 is there: 

![Annotation 2019-05-16 071736](https://user-images.githubusercontent.com/4986069/57815614-b701b980-77aa-11e9-993f-1fb390d78af5.jpg)

But I always get the error.

**Comment by SwimmingTiger** [comment]
Build with command (`.\tools\razzle.cmd` `bcz`) will get the same error:

![Annotation 2019-05-16 072646](https://user-images.githubusercontent.com/4986069/57815929-fc72b680-77ab-11e9-87d1-9befdcb76ab0.jpg)

Tried both `x64 Native Tools Command Prompt for VS 2017` and `x64 Native Tools Command Prompt for VS 2019`.

**Comment by binarycrusader** [comment]
I've just built the latest gate tip with the latest VS2017 without issue.  Can you provide screenshots showing what the Visual Studio Installer shows as installed?

For example, on my system:
![image](https://user-images.githubusercontent.com/541114/57824690-bb00ed80-7750-11e9-8467-0a08337f5937.png)

![image](https://user-images.githubusercontent.com/541114/57824729-e388e780-7750-11e9-9708-a14bc8c1ff49.png)

If you run "git status" from the command line in the source directory, does it show any changes?

Have you tried creating a new repository clone?

**Comment by oising** [comment]
Just an FYI @SwimmingTiger -- v142 requires v141, but this dependency is hidden from you and is automatically installed. I'd imagine v141 may have a dependency on parts of v140. So installing VS 2015 is not the correct interpretation here.

**Comment by binarycrusader** [comment]
See also #565.

**Comment by SwimmingTiger** [comment]
@binarycrusader I have the same components as you in VS2017, except IntelliTrace (the community version does not have this feature, this should not matter).

I also installed the same components as #565 in VS2019, but I still have this error.
This should be considered as a new issue. If it doesn't happen to you, maybe it only happens on Windows 10 version 1903 build 18895.1000?

![Screenshot (3)](https://user-images.githubusercontent.com/4986069/57833089-d7059d00-77eb-11e9-8ecd-e2c622c98d17.png)

**Comment by SwimmingTiger** [comment]
Eventually I solved the problem by "Repair" my VS2017. I noticed that a new directory `Microsoft.VCLibs.Desktop.110` was added to `C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs` (it didn't exist before. I didn't add or remove any components).

![Annotation 2019-05-16 070650](https://user-images.githubusercontent.com/4986069/57836173-bc82f200-77f2-11e9-8bed-09190355455f.jpg)

This repair also solved the same build problem in VS2019 (I didn't repair VS2019, just repaired VS2017). I guess these files are shared components of Visual Studio and were corrupted when I uninstalled VS2017 (at that time, I just wanted to keep VS2019) and were not reinstalled correctly after I reinstalled VS2017.
